### PR TITLE
Issue#89

### LINKED ISSUE #44: On_off_Controller with MotorDashboard plotted_variables='all' is not plotting anything
Running pi_series_omega_control.py with a MotorDashboard(plotted_variables='all', visu_period=1) and with an on_off_controller, is not plotting any progress on the diagrams.

Running the script with another controller or with a certain set of plotted_variables will work normally though.

### LINKED ISSUE #89: Final Dashboard Refactoring
Last open points for the dashboard:

- Cleanup of the code
- Easier parameterization (e.g. Plot widths)
- Minor changes in class hierarchy

### PR DESCRIPTION
**Pull request by atra94**
All suggestions from #89 have been introduced.

The documentation has been improved. A short usage guide has been added to the sphinx-docu. This might be useful for further components. So please, review is critically.

Closes #89
Closes #44

### CODE REVIEW

**Review by max-schenke — COMMENTED** [review]
No concerns upon visual code inspection, will try to test functionality later.
Is the constraint monitor now without unit tests?

**Review by max-schenke — APPROVED** [review]
Dashboard works on my machine, I think it looks better than before.
I am not sure whether we should write angles with pseudo unit "rad" since that is not an actual unit.
Maybe it would be better to drop the "rad" (rad / s => 1 / s, etc.).

The dashboard interface looks more structured now, which provides better usability.

On my system, when a dashboard is opened, the subplots are sometimes so small that their legend is too large for the plotting window. Maybe we could address that in the future.

Apart from these points I see no problem with this update. Thank you @atra94!

**Review by wallscheid — APPROVED** [review]
Also worked nicely on my machine. Except the small labeling issues already noted by @max-schenke regarding the unit depicting of an angular frequency, I do not have additional comments.

File: gym_electric_motor/core.py [hunk 1]
```diff
@@ -740,19 +681,32 @@ def on_reset_begin(self):
...
         """Gets called at the beginning of each step"""
         pass
     
-    def on_step_end(self):
+    def on_step_end(self, k, state, reference, reward, done):
         """Gets called at the end of each step"""        
         pass
     
     def on_close(self):
         """Gets called at the beginning on a close"""
```
Comment (wkirgsn) [inline gym_electric_motor/core.py:697]: ".. of a close"

File: gym_electric_motor/visualization/motor_dashboard.py [hunk 2]
```diff
@@ -1,140 +1,223 @@
...
+        self._episodic_plot_figure = None
+        self._interval_plot_figure = None
+
+        # Store the input data
+        self._state_plots = state_plots
+        self._action_plots = action_plots
+        self._reward_plot = reward_plot
+
+        # Separate the additional plots into StepPlots, EpisodicPlots and IntervalPlots
+        self._custom_step_plots = list(filter(lambda plot: isinstance(plot, TimePlot), list(additional_plots)))
```
Comment (wkirgsn) [inline gym_electric_motor/visualization/motor_dashboard.py:75]:
```suggestion
        self._custom_step_plots = [p for p in additional_plots if isinstance(p, TimePlot)]
```

the filter function does not seem necessary

File: gym_electric_motor/visualization/motor_dashboard_plots/state_plot.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        rg = env.reference_generator
+        # Save the index of the state.
+        self._state_idx = ps.state_positions[self._state]
+        # The maximal values of the state.
+        self._limits = ps.limits[self._state_idx]
+        self._state_space = ps.state_space.low[self._state_idx], ps.state_space.high[self._state_idx]
+        # Bool: if the state is referenced.
+        self._referenced = rg.referenced_states[self._state_idx]
+        if self._limits == self._state_space[1]:
+            self._normalized = False
```
Comment (wkirgsn) [inline gym_electric_motor/visualization/motor_dashboard_plots/state_plot.py:83]:
```suggestion
        self._normalized = self._limits != self._state_space[1]
```

**Review by wkirgsn — APPROVED** [review]
A reasonable refactoring. Almost nothing to criticize - well done!

**Pull request merged**